tests/test_dojo.py

```diff
@@ -76,21 +116,146 @@ def test_print_room(self):
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('john', 'swift')
+        self.assertEqual(res,None)
+
+    def test_reallocate_person_if_person_name_is_invalid(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('joe', 'swift')
+        self.assertEqual(res,None) 
+
+    def test_reallocate_person_to_livingspace_if_person_is_staff(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("kimani", "staff")
+        res = self.dojo.reallocate_person('kimani', 'swift')
+        self.assertEqual(res,None)        
+
+    def test_reallocate_to_same_room(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('joe', 'hogwarts')
+        self.assertEqual(res,None)  
+
+    def test_a_person_has_been_removed_from_a_room_after_reallocation(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.add_person('wick', 'fellow')
+        office = self.dojo.dojo_office[0]
+        members = office.members
+        self.assertEqual(len(members), 1)
+        self.dojo.create_room('php', 'office')
+        self.dojo.reallocate_person('wick', 'php')
+        office = self.dojo.dojo_office[0]
+        members = office.members
+        self.assertEqual(len(members), 0)
+
+    def test_if_a_room_doesnt_exist(self):
+        self.dojo.create_room('hogwarts','office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person("john", "fellow")
+        res = self.dojo.reallocate_person('joe', 'hogwarts3')
+        self.assertEqual(res,None) 
+
+        # test for load people from file --->
+    def test_load_people(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.create_room('swift', 'livingspace')
+        staff = self.dojo.staff
+        fellows = self.dojo.fellows
+        self.assertEqual(len(staff), 0)
+        self.assertEqual(len(fellows), 0)
+        self.dojo.load_people("sample")
+        self.assertEqual(len(staff), 3)
+        self.assertEqual(len(fellows), 4)
+
+        # tests for load and save data -->
+    def load_data(self):
+        self.dojo.create_room('hogwarts', 'office')
+        self.dojo.create_room('swift', 'livingspace')
+        self.dojo.add_person('burney', 'fellow', 'Y')
+        self.dojo.add_person('martial', 'staff')
+        return
+
+    def test_saves_state(self):
+        data = self.load_data()
+        create_db("databasefiles/test.db")
+        self.dojo.save_state('test')
+        # check if file is created
+        self.assertFalse(os.path.exists("databasefiles/test.db"))
```

Comment:
Checking if the file has been created isn't the best way to test this functionality. You don't know if it has the content you want. You could import sqlite3 (if it's an sqlite database) and assert that the columns contain the content you expect.